### PR TITLE
Convert node shutdown system property feature flag to setting

### PR DESCRIPTION
**Pull request by dakrone**
This converts the system property feature flag '`es.shutdown_feature_flag_enabled`' to a regular
non-dynamic node setting. This setting can only be set to 'true' on a snapshot build of
Elasticsearch (not a release build).

Relates to #70338

### CODE REVIEW

File: x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/ShutdownPlugin.java [hunk 1]
```diff
@@ -27,17 +29,24 @@
...
 public class ShutdownPlugin extends Plugin implements ActionPlugin {
 
-    public static final boolean SHUTDOWN_FEATURE_FLAG_ENABLED = "true".equals(System.getProperty("es.shutdown_feature_flag_enabled"));
+    public static final String SHUTDOWN_FEATURE_ENABLED_FLAG = "es.shutdown_feature_flag_enabled";
+    public static final Setting<Boolean> SHUTDOWN_FEATURE_ENABLED_FLAG_SETTING = Setting.boolSetting(
+        SHUTDOWN_FEATURE_ENABLED_FLAG,
+        false,
+        enabled -> {
+            if (enabled != null && enabled && Build.CURRENT.isSnapshot() == false) {
+                throw new IllegalArgumentException("shutdown plugin may not be enabled on a non-shutdown build");
```
Comment (gwbrown) [inline x-pack/plugin/shutdown/src/main/java/org/elasticsearch/xpack/shutdown/ShutdownPlugin.java:38]:
```suggestion
                throw new IllegalArgumentException("shutdown plugin may not be enabled on a non-snapshot build");
```

**Review by AthenaEryma — APPROVED** [review]
Left one minor suggestion. My only concern is: Are we sure that the tests will pass in release builds? Or do we need something like `assumeTrue` to prevent the integration tests from running? If we're good on that front, I'm good to merge this.

(admittedly depending on timing we may get rid of the feature flag before the next release is cut, but I'd rather be on the safe side)

**Comment by dakrone** [comment]
> Are we sure that the tests will pass in release builds? Or do we need something like assumeTrue to prevent the integration tests from running?

That's a good point. I think this might cause ES not to start for integration tests also, since we set it to true in `build.gradle` in a few places. I will investigate this.

**Comment by dakrone** [comment]
@elasticmachine update branch

**Comment by dakrone** [comment]
@gwbrown I think this should do the trick, can you take a look?

**Review by AthenaEryma — APPROVED** [review]
LGTM once CI is happy.

My only question is: Why don't we need `assumeTrue` or similar in [NodeShutdownIT](https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/shutdown/qa/multi-node/src/javaRestTest/java/org/elasticsearch/xpack/shutdown/NodeShutdownIT.java)? I'd assume that would fail unless the feature flag is enabled too. 

(If it turns out we do, no need for another round of review, that's a very minimal change)

**Comment by dakrone** [comment]
> Why don't we need assumeTrue or similar in NodeShutdownIT? I'd assume that would fail unless the feature flag is enabled too.

Yep, I think we need it there too, I'll add it to all those tests

**Pull request merged**